core/src/main/java/io/confluent/kafka/schemaregistry/metrics/MetricsContainer.java

```diff
@@ -174,21 +198,41 @@ private SchemaRegistryMetric getSchemaTypeMetric(String type, boolean isRegister
     }
   }
 
-  private static String getCommitId() {
-    final String defaultValue = "Unknown";
+  private static List<String> getMetricReporterConfig(SchemaRegistryConfig config) {
+    List<String> classes = new ArrayList<>(config.getList(
+            ProducerConfig.METRIC_REPORTER_CLASSES_CONFIG));
+    try {
+      if (!classes.contains(TELEMETRY_REPORTER_CLASS)) {
+        Class.forName(TELEMETRY_REPORTER_CLASS);
+        classes.add(TELEMETRY_REPORTER_CLASS);
+      }
+    } catch (ClassNotFoundException cnfe) {
+      // Ignore
+    }
+    return classes;
+  }
 
-    String fileName = "/schema-registry-app.properties";
-    try (InputStream propFile = MetricsContainer.class.getResourceAsStream(fileName)) {
-      if (propFile != null) {
-        Properties props = new Properties();
-        props.load(propFile);
-        return props.getProperty("application.commitId", defaultValue).trim();
-      } else {
-        log.error("Cannot find properties file");
+  private static MetricsReporter getTelemetryReporter(List<MetricsReporter> reporters) {
+    for (MetricsReporter reporter : reporters) {
+      if (reporter.getClass().getName().equals(TELEMETRY_REPORTER_CLASS)) {
+        return reporter;
       }
-    } catch (IOException e) {
-      log.warn("Cannot parse properties file", e);
     }
-    return defaultValue;
+    return null;
+  }
+
+  private static MetricsContext getMetricsContext(SchemaRegistryConfig config,
+                                                  String kafkaClusterId) {
+    Map<String, Object> metadata =
+            config.originalsWithPrefix(CommonClientConfigs.METRICS_CONTEXT_PREFIX);
+
+    String clusterId = String.format("%s-%s", kafkaClusterId,
+            config.getString(SchemaRegistryConfig.SCHEMAREGISTRY_GROUP_ID_CONFIG));
```

Comment:
For components that are tied to a specific kafka cluster, we split out the tags rather than combining them. i.e. for connect to we use the group id as the cluster.id, and we add a nother tag or the kafka cluster id.